java/native/HelloWorld/client/src/main/java/org/armonik/client/Main.java

```diff
@@ -0,0 +1,143 @@
+                                                                .build())
+                                                .build());
+
+                System.out.println(">> RESULTS RAW: ");
+                results.forEach(System.out::println);
+
+                List<TaskCreation> taskCreations = new ArrayList<>();
+                for (ResultRaw resultRaw : results) {
+                        TaskCreation taskCreation = TaskCreation
+                                        .newBuilder()
+                                        .setPayloadId(resultRaw.getResultId())
+                                        .addExpectedOutputKeys(resultRaw.getResultId())
+                                        .build();
+                        taskCreations.add(taskCreation);
+                }
+
+                // submit and get the taskId for the submitted task
+                String taskId = taskClient
+                                .submitTasks(SubmitTasksRequest.newBuilder()
+                                                .setSessionId(sessionId)
+                                                .setTaskOptions(taskOptions)
+                                                .addAllTaskCreations(taskCreations.stream()
+                                                                .<TaskCreation>map(taskCreation -> {
+                                                                        boolean taskOptionExist = taskCreation
+                                                                                        .getTaskOptions()
+                                                                                        .getMaxRetries() != Integer.MIN_VALUE;
+                                                                        if (!taskOptionExist) {
+                                                                                return taskCreation.toBuilder()
+                                                                                                .setTaskOptions(taskOptions)
+                                                                                                .build();
+                                                                        }
+                                                                        return taskCreation;
+                                                                }).toList())
+                                                .build())
+                                .getTaskInfosList()
+                                .get(0)
+                                .getTaskId();
+
+                System.out.println(">> Task ID: " + taskId);
+                EventClient eventClient = new EventClient(managedChannel);
+                List<String> resultsIds = results.stream().map(ResultRaw::getResultId).toList();
+
+                // getting events
+
+                List<EventSubscriptionResponseRecord> records = eventClient.getEventResponseRecords(sessionId,
```

Comment:
That's not for today but what if you have to want for 1 million data ? you'll get a lot of records in your list. I hope your machine has enough RAM. I think you should not return those.

Reply:
you are right. The idea here was to show how it works